pulsar-client-api/src/main/java/org/apache/pulsar/client/api/AuthenticationDataProvider.java

```diff
@@ -107,4 +123,11 @@ default String getCommandData() {
         return null;
     }
 
+    /**
+     * Refresh Authentication Data.
+     * 
+     */
+    default void refresh() {
```

Comment:
But shouldn't this be dealt internally by implementations? Why should we triggering it from outside? 

For example: when reading token credential you have different ways to refresh the token data (eg: passing a `Supplier<String>` or by refreshing the token file). 

The implementation will have more information about when it has to refresh, so it would be better to leave the decision inside there.

My point is that TLS plugin should do the caching in its own implementation, for example by checking the last modified of cert file when client lib calls `getTlsCertificates()`